modin/logging/logger_decorator.py

```diff
@@ -103,11 +103,13 @@ def decorator(obj):
         elif isinstance(obj, staticmethod):
             return staticmethod(decorator(obj.__func__))
 
+        assert isinstance(modin_layer, str)
```

Comment:
If we assert that `modin_layer` is a string, why does the typing allow it not to be a string?

Reply:
There's a bit of complexity here. Basically `enable_logging` returns a function since it's a decorator. If you see in the top of the function, we check to see whether `modin_layer` is a string or not. If it's not, then we call `enable_logging()(modin_layer)` which is basically doing `decorator(modin_layer)`. We can see in the implementation of `decorator` that we check to see what the type of `obj` is (and we claim earlier that it can be a string, type, or callable function). We check for these cases but I think `mypy` isn't convinced that we can get away with calling string methods on the union, so I added an assert for sanity. Does that make sense?

Reply:
mypy threw this error specifically:
```
modin/logging/logger_decorator.py:106:32: error: Item "function" of "Union[str, Callable[..., Any], Type[Any]]" has no attribute "upper"  [union-attr]
modin/logging/logger_decorator.py:107:30: error: Item "function" of "Union[str, Callable[..., Any], Type[Any]]" has no attribute "upper"  [union-attr]
```

Reply:
@mvashishtha would it make sense to add a string to give more context on a potential failure on this assert here?

Reply:
Yeah, I think a short comment would be good.

Reply:
I would advise against using bare `assert`-s, I think it's better to accompany each such statement with a proper assertion message which would double-serve as a comment